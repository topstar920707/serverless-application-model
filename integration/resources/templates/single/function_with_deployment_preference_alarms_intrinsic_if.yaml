Conditions:
  MyCondition:
    Fn::Equals:
      - true
      - false
Resources:
  MyLambdaFunction:
    Type: "AWS::Serverless::Function"
    Properties:
      CodeUri: ${codeuri}
      Handler: hello.handler
      Runtime: python3.8
      AutoPublishAlias: live
      DeploymentPreference:
        Type: Linear10PercentEvery3Minutes
        Alarms:
          Fn::If:
            - MyCondition
            - - Alarm1
              - Alarm2
              - Alarm3
            - - Alarm1
              - Alarm5
