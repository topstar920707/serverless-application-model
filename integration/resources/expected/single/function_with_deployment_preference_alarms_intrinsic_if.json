[
  {
    "LogicalResourceId": "ServerlessDeploymentApplication",
    "ResourceType": "AWS::CodeDeploy::Application"
  },
  {
    "LogicalResourceId": "MyLambdaFunctionRole",
    "ResourceType": "AWS::IAM::Role"
  },
  {
    "LogicalResourceId": "CodeDeployServiceRole",
    "ResourceType": "AWS::IAM::Role"
  },
  {
    "LogicalResourceId": "MyLambdaFunction",
    "ResourceType": "AWS::Lambda::Function"
  },
  {
    "LogicalResourceId": "MyLambdaFunctionDeploymentGroup",
    "ResourceType": "AWS::CodeDeploy::DeploymentGroup"
  },
  {
    "LogicalResourceId": "MyLambdaFunctionVersion",
    "ResourceType": "AWS::Lambda::Version"
  },
  {
    "LogicalResourceId": "MyLambdaFunctionAliaslive",
    "ResourceType": "AWS::Lambda::Alias"
  }
]